isce2_topsapp/localize_burst.py

```diff
@@ -0,0 +1,315 @@
+        base_path: Path = Path('.'),
+        download_strategy: str = 'single_burst') -> Path:
+    """Creates this file structure:
+    SLC.SAFE/
+    ├── manifest.safe
+    ├── measurement/
+    │   └── burst.tif
+    └── annotation/
+        ├── annotation.xml
+        └── calbiration/
+            ├── calibration.xml
+            └── noise.xml
+    """
+    safe_path = base_path / burst.safe_name
+    annotation_path = safe_path / 'annotation'
+    calibration_path = safe_path / 'annotation' / 'calibration'
+    measurement_path = safe_path / 'measurement'
+    paths = [annotation_path, calibration_path, measurement_path]
+    for p in paths:
+        if not p.exists():
+            p.mkdir(parents=True)
+
+    et_args = {'encoding': 'UTF-8', 'xml_declaration': True}
+
+    ET.ElementTree(burst.annotation).write(annotation_path / burst.annotation_name, **et_args)
+    ET.ElementTree(burst.calibration).write(calibration_path / burst.calibration_name, **et_args)
+    ET.ElementTree(burst.noise).write(calibration_path / burst.noise_name, **et_args)
+    ET.ElementTree(burst.manifest).write(safe_path / 'manifest.safe', **et_args)
+
+    if download_strategy == 'single_burst':
+        burst_params = BurstParams(
+            safe_url=burst.safe_url, image_number=burst.image_number, burst_number=burst.burst_number
+        )
+        download_geotiff(
+            asf_session, burst_params, measurement_path / burst.measurement_name
+        )
+    elif download_strategy == 'surrounding_burst':
+        n_bursts = len(burst.annotation.find('.//burstList'))
+        names = {
+            'burst_pre.tiff': burst.burst_number - 1,
+            burst.measurement_name: burst.burst_number,
+            'burst_post.tiff': burst.burst_number + 1,
+        }
+        names = {k: v for k, v in names.items() if 0 < v <= n_bursts}
+        for n in names:
+            burst_params = BurstParams(
+                safe_url=burst.safe_url, image_number=burst.image_number, burst_number=names[n]
+            )
+            download_geotiff(asf_session, burst_params, measurement_path / n)
+    elif download_strategy == 'swath':
+        download_swath(
+            burst.safe_url,
+            measurement_path,
+            burst.measurement_name,
+        )
+    else:
+        raise NotImplementedError(f'Download strategy {download_strategy} is not implemented, check spelling.')
+
+    return safe_path
+
+
+# TODO currently only validated for descending orbits
+def get_region_of_interest(
+        poly1: geometry.Polygon,
+        poly2: geometry.Polygon,
+        asc: bool = True) -> Tuple[float, float, float, float]:
+    bbox1 = geometry.box(*poly1.bounds)
+    bbox2 = geometry.box(*poly2.bounds)
+    intersection = bbox1.intersection(bbox2)
+    bounds = intersection.bounds
+
+    x, y = (0, 1) if asc else (2, 1)
+    roi = geometry.Point(bounds[x], bounds[y]).buffer(0.005)
+    minx, miny, maxx, maxy = roi.bounds
+    return minx, miny, maxx, maxy
```

Comment:
Any reason to not just return `roi.bounds` and document the order they are returned in?

Reply:
No, not really.